Backend/Tournament_Admin/TournamentAdminService/src/main/java/TournamentAdminService/service/MatchupService.java

```diff
@@ -0,0 +1,87 @@
+package TournamentAdminService.service;
+
+import TournamentAdminService.model.Matchup;
+import TournamentAdminService.repository.MatchupRepository;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.List;
+
+/**
+ * Service class for managing matchup-related operations.
+ * This class provides methods to update, delete, and retrieve matchups and participants.
+ */
+@Service
+public class MatchupService {
+    @Autowired
+    private MatchupRepository matchupRepository;
+
+    /**
+     * Creates a new game result for a tournament matchup.
+     *
+     * @param playerWon the player who won the game
+     * @param tournamentId the ID of the tournament
+     * @param roundNum the round number
+     */
+    @Transactional
+    public void createGameResult(String playerWon, String tournamentId, Integer roundNum) {
+        // Input validation
+        if (tournamentId == null || roundNum == null || playerWon == null) {
```

Comment:
not sure if Java validates empty Strings as null, i think can consider the [isEmpty()](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#isEmpty--) method as well/possibly move to another util file if this validation is reused? but generally ok

in terms of the JSON response rather than a null value it could be 

```json
{"playerWon": ""} // not null value technically
```

Reply:
if uw i have a func in ValidationUtil thats checks if a string is valid or not ah if not throws IllegalArgumentException

```
    /**
     * Validates that a string is not null or empty.
     *
     * @param value     The string to validate.
     * @param fieldName The name of the field being validated (for error messages).
     * @throws IllegalArgumentException if the string is null or empty.
     */
    public static void validateNotEmpty(String value, String fieldName) {
        if (value == null || value.trim().isEmpty()) {
            throw new IllegalArgumentException(fieldName + " must not be null or empty.");
        }
    }
    ```

Reply:
done